guides/ansible_engine/5-role/README.md

```diff
@@ -0,0 +1,221 @@
+# Exercise 1.5 - Roles: Making your playbooks reusable
+
+While it is possible to write a playbook in one file as we've done throughout this workshop, eventually you’ll want to reuse files and start to organize things.
+
+Ansible Roles is the way we do this.  When you create a role, you deconstruct your playbook into parts and those parts sit in a directory structure.  "Wha??  You mean that seemingly useless [best practice](http://docs.ansible.com/ansible/playbooks_best_practices.html) you mentioned in exercise 1.2?".  Yep, that one.
+
+For this exercise, you are going to take the playbook you just wrote and refactor it into a role.  In addition, you'll learn to use Ansible Galaxy.
+
+Let's begin with seeing how your apache-basic-playbook will break down into a role.
+
+![Role directory structure](roledir_1.png)
+
+Fortunately, you don't have to create all of these directories and files by hand.  That's where Ansible Galaxy comes in.
+
+## Section 1: Using Ansible Galaxy to initialize a new role
+
+Ansible Galaxy is a free site for finding, downloading, and sharing roles.  It's also pretty handy for creating them which is what we are about to do here.
+
+
+### Step 1:
+
+Navigate to your `apache-basic-playbook` project.
+
+```bash
+cd ~/apache-basic-playbook
+```
+
+
+### Step 2:
+
+Create a directory called `roles` and `cd` into it.
+
+```bash
+mkdir roles
+cd roles
+```
+
+
+### Step 3:
+
+Use the `ansible-galaxy` command to initialize a new role called `apache-simple`.
+
+```bash
+ansible-galaxy init apache-simple
+```
+
+Take a look around the structure you just created.  It should look a lot like Figure 1 above.  However, we need to complete one more step before moving onto section 2.  It is Ansible best practice to clean out role directories and files you won't be using.  For this role, we won't be using anything from `files`, `tests`.
+
+
+### Step 4:
+
+Remove the `files` and `tests` directories
+
+```bash
+cd ~/apache-basic-playbook/roles/apache-simple/
+rm -rf files tests
+```
+
+
+## Section 2: Breaking Your `site.yml` Playbook into the Newly Created `apache-simple` Role
+
+
+In this section, we will separate out the major parts of your playbook including `vars:`, `tasks:`, `template:`, and `handlers:`.
+
+### Step 1:
+
+Make a backup copy of `site.yml`, then create a new `site.yml`.
```

Comment:
ugh. really? I realize stepping students thru version controlling the content is out of scope even though it's best practice. This is a bit too crude. Thinking this exercise should have them start a new project directory `~/apache-role/` and copy things from apache-simple-playbook directory as needed. That would be consistent with the workshops.

Reply:
I very much prefer to use a new directory here, called `ansible-role`. Otherwise, things get mixed up way too much (ansible-basic-playbook, ansible-simple-playbook, now also ansible-simple, etc.).
I will rewrite to base this chapter on `ansible-role`.

Reply:
sounds like a plan then @liquidat. with so much moving around we'll probably end up merging, see more rough edges and taking another pass at it anyways.